### PR TITLE
Standardize on configuration files and use of Tranquilizer.

### LINKED ISSUE #103: Config file should be usable by Core API
Instead of just Server and Kafka. This should make it easier for people to use whichever module makes more sense for them, without changing a bunch of things.

### PR DESCRIPTION
**Pull request by gianm**
- Introduce TranquilityConfig.read and DruidBeams.fromConfig.
- Switch all modules from simple tranquilizer to Tranquilizer + flush.
- Switch Java/Scala code examples to Tranquilizer + flush.
- Adjust default config files to have more consistent formatting.
- Allow 'dataSources' to be a list instead of a map in config files.

Also:
- Include full date in bundled logback.xml

Fixes #103

### CODE REVIEW

File: kafka/src/main/java/com/metamx/tranquility/kafka/KafkaMain.java [hunk 1]
```diff
@@ -82,27 +81,29 @@ public void run() throws InterruptedException
...
     final Properties kafkaProperties = new Properties();
-    for (Map.Entry<String, Object> entry : globalProperties.entrySet()) {
-      String key = entry.getKey();
-      log.info("%s = %s", key, globalProperties.get(key).toString());
-      if (key.startsWith("kafka.")) {
-        kafkaProperties.setProperty(key.replaceFirst("kafka.", ""), globalProperties.get(key).toString());
+    for (String propertyName : config.globalConfig().properties().stringPropertyNames()) {
+      if (propertyName.startsWith("kafka.")) {
+        kafkaProperties.setProperty(
+            propertyName.replaceFirst("kafka\\.", ""),
```
Comment (dclim) [inline kafka/src/main/java/com/metamx/tranquility/kafka/KafkaMain.java]: Nice catch

File: kafka/src/main/java/com/metamx/tranquility/kafka/writer/TranquilityEventWriter.java [hunk 2]
```diff
@@ -90,31 +92,57 @@ public void send(byte[] message) throws InterruptedException
...
+      final long rejectedLogCount = rejectedLogCounter.incrementAndGet();
+      if (rejectedLogCount <= 5
           || (rejectedLogCount <= 100 && rejectedLogCount % 10 == 0)
           || rejectedLogCount % 100 == 0) {
         log.debug(e, "%d message(s) failed to parse as JSON and were rejected", rejectedLogCount);
       }
-      if (dataSourceConfig.config().reportDropsAsExceptions()) {
-        Throwables.propagate(e);
+      if (dataSourceConfig.propertiesBasedConfig().reportDropsAsExceptions()) {
+        failedCounter.incrementAndGet();
```
Comment (dclim) [inline kafka/src/main/java/com/metamx/tranquility/kafka/writer/TranquilityEventWriter.java]: Perhaps failedCounter should be incremented regardless if we throw an exception or not.

Comment (gianm) [inline kafka/src/main/java/com/metamx/tranquility/kafka/writer/TranquilityEventWriter.java]: that's reasonable and in line with what the block below is doing.

**Comment by gianm** [comment]
I like cats

File: core/src/test/resources/example.json [hunk 3]
```diff
@@ -0,0 +1,76 @@
+{
+  "dataSources": {
```
Comment (fjy) [inline core/src/test/resources/example.json]: should this be an array?

File: core/src/test/resources/example.json [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+              "fieldName": "delta"
+            }
+          ],
+          "granularitySpec": {
+            "type": "uniform",
+            "segmentGranularity": "DAY",
+            "queryGranularity": "NONE"
+          }
+        },
+        "tuningConfig": {
```
Comment (fjy) [inline core/src/test/resources/example.json]: do these match druid tuningConfig params?

**Comment by fjy** [comment]
:+1: on the docs and configuration

File: core/src/test/resources/example.json (hunk 4 above)
Comment (gianm) [inline core/src/test/resources/example.json]: They do although currently tranquility doesn't support all of them. see also #116

File: core/src/test/resources/example.json (hunk 3 above)
Comment (gianm) [inline core/src/test/resources/example.json]: ah yeah good point.

Comment (gianm) [inline core/src/test/resources/example.json]: fixed

File: core/src/main/scala/com/metamx/tranquility/druid/DruidBeams.scala [hunk 5]
```diff
@@ -277,14 +267,37 @@ object DruidBeams
   class Builder[EventType] private[druid](private[tranquility] val config: BuilderConfig[EventType])
   {
     /**
-      * Provide a CuratorFramework instance that will be used for Tranquility's internal coordination. Required.
+      * Provide a CuratorFramework instance that will be used for Tranquility's internal coordination. Either this
+      * or "curatorFactory" is required. Whichever one is provided later will win.
+      *
+      * If you provide Curator this way (as opposed to "curator") then you should start and stop it yourself. You can
```
Comment (dclim) [inline core/src/main/scala/com/metamx/tranquility/druid/DruidBeams.scala]: Should this say 'as opposed to "curatorFactory"'?

Comment (gianm) [inline core/src/main/scala/com/metamx/tranquility/druid/DruidBeams.scala]: yes, definitely.

**Comment by dclim** [comment]
Looks good to me, standardized configs is awesome.
:+1:

**Pull request merged**